src/main/scala/com/imaginea/activegrid/core/models/Neo4jRepository.scala

```diff
@@ -0,0 +1,250 @@
+        createNode(label)
+      }
+    }
+  }
+
+  /**
+    * Returns all the nodes given with the node label
+    *
+    * @param label
+    * @return
+    */
+  def getNodesByLabel(label: String): List[Node] = withTx { neo =>
+    getAllNodesWithLabel(label)(neo).toList
+  }
+
+  /**
+    * Returns values as a map for the given keys from node
+    *
+    * @param node
+    * @param keys
+    * @return
+    */
+  def getProperties(node: Node, keys: String*): Map[String, Any] = withTx { neo =>
+    val map: scala.collection.mutable.Map[String, AnyRef] = scala.collection.mutable.Map()
+    keys.foreach(key => {
+      try {
+        val value = node.getProperty(key)
+        logger.debug(s" (${key}) --> (${node.getProperty(key)}) ")
+        map += ((key, value))
+      } catch {
+        case ex: Throwable => logger.warn(s"failed to get values for the key ${key}")
+      }
+    })
+    map.toMap
+  }
+
+  /**
+    * Finds the node with given node and property information
+    *
+    * @param label
+    * @param propertyName
+    * @param propertyValue
+    * @tparam T
+    * @return
+    */
+  def getEntity[T <: BaseEntity : Manifest](label: String, propertyName: String, propertyValue: Any): T = withTx { neo =>
+    val nodes = findNodesByLabelAndProperty(label, propertyName, propertyValue)(neo)
+    nodes.map(_.toCC[T].get).toList.head
+  }
+
+  /**
+    * Deletes the Incoming / out going relations along with the node
+    *
+    * @param nodeId
+    * @return
+    */
+  def deleteChildNode(nodeId: Long): Option[Boolean] = withTx { implicit neo =>
+
+    // get realtion with parent  (incoming relations)
+    // delete incoming
+    // delete node
+    logger.debug(s"trying to delete node with id ${nodeId}")
+    val node = getNodeById(nodeId)
+    val incomingRelations = getRelationships(node, Direction.INCOMING)
+    incomingRelations.foreach(incomingRelation => {
+      logger.debug(s"Deleting incoming relation ${incomingRelation}  -- ${incomingRelation.getType}")
+      incomingRelation.delete()
+    })
+
+    deleteEntity(nodeId)
+    Some(true)
+  }
+
+  /**
+    * deletes the node with the given label and property details
+    * deletes all out going relations from the node as well
+    *
+    * @param nodeId
+    */
+  def deleteEntity(nodeId: Long): Unit = withTx { implicit neo =>
+    val node = findNodeById(nodeId).get
+    val relations = getRelationships(node, Direction.OUTGOING)
+    logger.debug(s"found relations for node ${nodeId} - relations ${relations}")
+
+    relations.foreach(outRelation => {
+      logger.debug(s"deleting out relation ${outRelation}  -- ${outRelation.getType}")
+      outRelation.delete()
+    })
+
+    logger.debug(s"finally deleting node ${node}")
+    node.delete
+  }
+
+  /**
+    * Returns the node with given id if not found throws not found excpetion
+    *
+    * @param id
+    * @return
+    */
+  def findNodeById(id: Long): Option[Node] = withTx { implicit neo =>
```

Comment:
Option already implies Node might not be there, why are we throwing instead of using None?